pallets/scheduler/src/lib.rs

```diff
@@ -435,24 +450,26 @@ pub mod pallet {
 					continue;
 				}
 
-				let sender = ensure_signed(
-					<<T as Config>::Origin as From<T::PalletsOrigin>>::from(s.origin.clone())
-						.into(),
-				)
-				.unwrap();
-
-				// // if call have id it was be reserved
-				// if s.maybe_id.is_some() {
-				// 	let _ = T::CallExecutor::pay_for_call(
-				// 		s.maybe_id.unwrap(),
-				// 		sender.clone(),
-				// 		call.clone(),
-				// 	);
-				// }
-
-				// Execute transaction via chain default pipeline
-				// That means dispatch will be processed like any user's extrinsic e.g. transaction fees will be taken
-				let r = T::CallExecutor::dispatch_call(sender, call.clone());
+				let scheduled_origin =
+					<<T as Config>::Origin as From<T::PalletsOrigin>>::from(s.origin.clone());
+				let ensured_origin =
+					T::ScheduleOrigin::ensure_origin(scheduled_origin.into()).unwrap();
+
+				let r;
+				match ensured_origin {
+					ScheduledEnsureOriginSuccess::Root => {
+						r = Ok(call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into()));
+					}
+					ScheduledEnsureOriginSuccess::Signed(sender) => {
+						// Execute transaction via chain default pipeline
+						// That means dispatch will be processed like any user's extrinsic e.g. transaction fees will be taken
+						r = T::CallExecutor::dispatch_call(Some(sender), call.clone());
+					}
+					ScheduledEnsureOriginSuccess::Unsigned => {
```

Comment:
Unsigned = inherents? Не должен ли тут использоваться код аналогичный Root?

Reply:
С сайта Substrate: `Inherent transactions—sometimes referred to as inherents—are a special type of unsigned transaction` -- [unsigned txs](https://docs.substrate.io/fundamentals/transaction-types/#unsigned-transactions).

Может быть и не Root, судя по тексту на сайте. Может быть что-то специфичное.

Reply:
И обрабатываться максимально специфично: `Because unsigned transactions require custom validation, this transaction type consumes more resources than a signed transaction`

Reply:
Валидация нужна специфичная, т.к inherents по сети не распространяются
И в данном случае мне кажется лишним запускать SignedExtensions для них

Reply:
Может, закомментированный поддержку unsigned транзакций, пока мы ими не пользуемся?
Вернём когда потребуются, и по гарантиям будет понятнее